src/definition/scheduler/IProcessor.ts

```diff
@@ -1,10 +1,42 @@
 import {IHttp, IModify, IPersistence, IRead} from '../accessors';
 
+/** Represents a processor that is used by the scheduler methods */
 export interface IProcessor {
+    /** The processor's identifier */
     id: string;
+    /** The function that will be run on a given scheudle */
     processor: (jobContext: IJobContext, read: IRead, modify: IModify, http: IHttp, persis: IPersistence) => Promise<void>;
+    /** If provided, the processor will be configured with the setting as soon as it gets registered */
+    startupSetting?: StartupSetting;
 }
 
+/** The `data` object provided to the processor during the registering process */
 export interface IJobContext {
     [key: string]: any;
 }
+
+type StartupSetting = IOnetimeStartup | IRecurringStartup;
+
+/**
+ * Processor setting for running once after being registered
+ */
+export interface IOnetimeStartup {
+    type: StartupType.ONETIME;
+    when: string;
+}
+
+/**
+ * Processor setting for running recurringly after being registered
+ */
+export interface IRecurringStartup {
```

Comment:
This interface should support a `data` property, similar to the `IRecurringSchedule` interface, to support passing arguments over to the processor

Reply:
Done!